code/datums/quirks/negative.dm

```diff
@@ -1075,3 +1074,123 @@
 	if(!IS_ORGANIC_LIMB(old_limb))
 		cybernetics_level--
 		update_mood()
+
+/datum/quirk/decrepit
+	name = "Decrepit"
+	desc = "You require a cane like object in your hand for mobility."
+	icon = "person-cane"
+	value = -4
+	hardcore_value = 4
+	gain_text = "<span class='danger'>You feel old!</span>"
+	lose_text = "<span class='notice'>You feel young again.</span>"
+	medical_record_text = "Patient suffers from a lack of mobility requiring a cane for support."
+	mail_goodies = list(/obj/item/cane)
+	var/datum/weakref/left_cane
+	var/datum/weakref/right_cane
+
+/datum/quirk/decrepit/add_unique()
+	var/mob/living/carbon/human/human_holder = quirk_holder
+
+	var/turf/holder_turf = get_turf(human_holder)
+	var/obj/item/cane/spawn_cane = new /obj/item/cane(holder_turf)
+	human_holder.put_in_hands(spawn_cane)
+
+	var/obj/item/left_hand_item = human_holder.held_items[1]
+	var/obj/item/right_hand_item = human_holder.held_items[2]
+	var/is_cane_left_hand = istype(left_hand_item) && HAS_TRAIT(left_hand_item, TRAIT_CANE_TOOL)
+	var/is_cane_right_hand = istype(right_hand_item) && HAS_TRAIT(right_hand_item, TRAIT_CANE_TOOL)
```

Comment:
man just loop through held_items why do this

Reply:
Was trying to avoid making this a `process()` quirk since it's costly to run every tick.  Decided to tie all the code to the movement signal instead with https://github.com/tgstation/tgstation/pull/71470/commits/f3f3eadc10f674e64f41057130ed7eefa74ec715